01week/rockPaperScissors.js

```diff
@@ -7,26 +7,65 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+/*Whiteboard plan:
+Rules-
+Scissors beats paper | paper beats rock | rock beats scissors | both hands equal is a tie
 
-function rockPaperScissors(hand1, hand2) {
+1. Create a function that makes inputs all lower case and removes whitespace (and store them in variables) PLUS checks that user inputs are valid. If everything passes, move on to #2 ELSE display "invalid entry"
 
-  // Write code here
+-use toLowerCase() and trim() to force inputs to lower case and to remove whitespace
+-use includes() to verify that both inputs exactly match the valid options which are stored in an array
+-if both inputs are valid, move onto the next function, if both are not valid, display "invalid input"
 
+2. Create function that determines if user1 wins, user2 wins or it's a tie. Inside this function, create an if else/if else conditional for a tie first, then test for user1 win, ELSE user2 wins
+
+-a tie can be detrmined if both inputs are equal
+-a hand1 win can be detrmined by comparing both inputs and accounting for 3 separate ways hand1 could win.
+-a hand2 win does not need to be directly determined, since if a tie or hand1 win has not occured, it's the only other possible outcome.
+3. return the result of the game (hand1 win, hand2 win, tie)
+
+*/
+
+
+const isItValid=(hand1, hand2)=> {//check to see if answers are valid before comparing
+  const possibleHands = ["rock", "paper", "scissors"];//array of valid inputs
+  const newHand1 = hand1.toLowerCase().trim();//remove caps and white space
+  const newHand2 = hand2.toLowerCase().trim();//remove caps and white space
+
+  if(possibleHands.includes(newHand1) === true && possibleHands.includes(newHand2) === true) {
+    return rockPaperScissors(newHand1, newHand2);//run rockPaperScissors() only if both entries are valid
+  }else{
+    return 'Inavlid entry!! Please type rock, paper or scissors.'//else tell user of invalid entry(ies)
+  }
+}
+
+
+const rockPaperScissors=(newHand1, newHand2)=> {//this runs ONLY if isItValid() passes
+  if (newHand1 === newHand2) {//first check for a tie
+    return 'It`s a tie!!'//if true, it's a tie, if false move on
+  } else if (newHand1 === 'rock' && newHand2 === 'scissors' ||//if no tie, check if hand1 is a winner
```

Comment:
you don't need the third 'newHand1 === 'paper', it can be simplified to `|| newHand2 ==='rock'`

Reply:
Actually..I don't think that is true. Using your suggestions, hand1 could ===scissors, and hand2 could ===rock and it would report a hand1 win, but hand2 should win in that case.